scala-package/core/src/main/scala/org/apache/mxnet/ExecutorManager.scala

```diff
@@ -395,17 +395,17 @@ private[mxnet] object ExecutorManager {
  * @param paramNames Names of all trainable parameters.
  * @param ctx List of devices for training (data parallel)
  * @param slices Describes how the data parallel splits data into different devices.
- * @param providedData training data shapes
- * @param providedLabel training label shapes
+ * @param providedDataDesc training data descriptions
+ * @param providedLabelDesc training label descriptions
  * @param sharedGroup: DataParallelExecutorGroup
  *                   An existing executor group, if to share parameters with it.
  *
  */
 private class DataParallelExecutorGroup private(sym: Symbol,
                                 argNames: IndexedSeq[String], paramNames: Set[String],
                                 ctx: Array[Context], private val slices: Array[(Int, Int)],
-                                providedData: Map[String, Shape],
-                                providedLabel: Map[String, Shape],
+                                providedDataDesc: IndexedSeq[DataDesc],
```

Comment:
Why we are making this changes?

Reply:
Since we are deprecating provideData and the usages of Map[String, Shape], we should attempt to do so universally to keep the Scala API simple. Additionally, all of the methods/constructors that were designed based on the fact that they would call provideData should be deprecated as they would otherwise throw a warning due to calling the deprecated methods